### PR TITLE
Improve code coverage

### PR DESCRIPTION
**Pull request by adejanovski**
Use an in process grpc server in ITs to get proper coverage stats.
Remove dead code as the grpc/restore.py script isn't used anywhere.
Display the number of files that were protected from deletion as they're still within the grace period and add a flag to override this grace period (+ documentation).

### CODE REVIEW

File: medusa/purge.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def backups_to_purge_by_count(backups, max_backup_count):
 #     return list(filter(lambda backup: backup.name = backup_name, backups)) or list()
 
 
-def purge_backups(storage, backups, backup_grace_period_in_days):
+def purge_backups(storage, backups, backup_grace_period_in_days, local_fqdn):
```
Comment (jeffbanks) [inline medusa/purge.py:86]: What is the primary reason for adding `local_fqdn` for purge?   Wondering if part of code coverage or something else?

**Review by jeffbanks — COMMENTED** [review]
Overall, looking like some great changes.  Just a few comments and questions.

File: medusa/purge.py (hunk 1 above)
Comment (adejanovski) [reply medusa/purge.py:86]: Good point. It actually shouldn't have been part of this PR as it fixes an issue related to the introduction of the backup grace period which could leave some files on disk which still need purging even if no backup was deleted for this round.

File: medusa/purge.py [hunk 2]
```diff
@@ -92,24 +92,35 @@ def purge_backups(storage, backups, backup_grace_period_in_days):
...
 
     for backup in backups:
         (purged_objects, purged_size) = purge_backup(storage, backup)
         nb_objects_purged += purged_objects
         total_purged_size += purged_size
         fqdns.add(backup.fqdn)
 
+    if len(fqdns) == 0:
+        # If we didn't purge any backup, we still want to cleanup obsolete files for the local node
+        fqdns.add(local_fqdn)
```
[the remark this answers is not in the document]
Comment (adejanovski) [reply medusa/purge.py:105]: The value will always be present as it defaults to the local node IP/hostname.

File: tests/backup_cluster_test.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    ERROR = 1
+
+
+class BackupClusterTest(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def setUp(self):
+        self.config = self._build_config_parser()
```
[the remark this answers is not in the document]
Comment (adejanovski) [reply tests/backup_cluster_test.py:40]: cool 😊

File: tests/integration/features/integration_tests.feature [hunk 4]
```diff
@@ -796,4 +797,42 @@ Feature: Integration tests
...
+        When I delete the backup named "first_backup"
+        Then I cannot see the backup named "first_backup" when I list the backups
+
+        @local
+        Examples: Local storage
+        | storage           | client encryption |
+        | local      |  with_client_encryption |
+    
+    @21
+    Scenario Outline: Create an corrupt backup, verify it and delete it
```
[the remark this answers is not in the document]
Comment (adejanovski) [reply tests/integration/features/integration_tests.feature:823]: Good catch, will fix it.

[the remark this answers is not in the document]
Comment (adejanovski) [reply tests/integration/features/integration_tests.feature:823]:
```suggestion
    Scenario Outline: Create a corrupt backup, verify it and delete it
```

**Comment by adejanovski** [comment]
@jeffbanks could you check our conversations here when you have some time and tell me if you're ok with the changes?

**Review by jeffbanks — APPROVED** [review]
Nice work!  The code coverage is really starting to pick up.

**Comment by maxbrunet** [comment]
> Remove dead code as the grpc/restore.py script isn't used anywhere.

It is used for restore in Kubernetes, now broken with v0.12.0 😅

https://github.com/thelastpickle/cassandra-medusa/blob/e7de031389947562a32ef98e2c581abad7a5ec9d/k8s/docker-entrypoint.sh#L44

**Pull request merged**